transaction/transaction.go

```diff
@@ -73,6 +78,106 @@ func NewTxInput(hash []byte, index uint32) *TxInput {
 	}
 }
 
+type AddIssuanceArg struct {
+	Precision           uint
+	Contract            *IssuanceContract
+	AssetAmount         float64
+	TokenAmount         float64
+	AssetAddress        string
+	TokenAddress        string
+	ReissuanceTokenFlag uint
+	Net                 network.Network
+}
+
+//AddIssuance adds non-confidential issuance to the transaction
+func (tx *Transaction) AddIssuance(arg AddIssuanceArg) (uint32, error) {
+	if len(tx.Inputs) == 0 {
+		return 0, errors.New("transaction must contain at least one input ")
+	}
+
+	if arg.AssetAmount == 0 {
+		return 0, errors.New("asset amount must be greater then 0")
+	}
+
+	if arg.AssetAddress == "" {
+		return 0, errors.New("asset address must be provided")
+	}
+
+	if arg.TokenAmount > 0 && arg.TokenAddress == "" {
+		return 0, errors.New("token address must be provided")
+	}
+
+	issuanceExt, err := NewTxIssuance(arg.AssetAmount, arg.TokenAmount,
+		arg.Precision)
+	if err != nil {
+		return 0, err
+	}
+
+	var inputIndex uint32
+	var inputHash []byte
+	for i, input := range tx.Inputs {
```

Comment:
considered the updates made to task, we should range over `u.Pset.UnsignedTx.Inputs`

Reply:
Done